### ISSUE TITLE
install.sh - Require gstreamer1.0-plugins-good on RPI4

### ISSUE THREAD

**Issue by ZeroErrors** [openDsh/dash#35]
# Issue

## Hardware

| Device             | Connection Method | Screen                       | Mobile Device     | Android Auto Version |
| ------------------ | ----------------- | ---------------------------- | ----------------- | -------------------- |
| Raspberry Pi 4 4GB | USB Cable         | Official Raspberry 7" Screen | Google Pixel XL | v5.9.604634-release          |

Installation Method: `install.sh`

## Description of problem:
When connecting Android device the projection would be solid white, same as described in issue https://github.com/openDsh/dash/issues/30.

I attempted disabling RTAudio and changing various other OpenAuto settings but none solved the problem.

After messing around with various things I found the section of code in openauto [GSTVideoOutput.cpp#L46-L64](https://github.com/openDsh/openauto/blob/develop/openauto/Projection/GSTVideoOutput.cpp#L46-L64) that controlled the codec used.
I then ran `gst-inspect-1.0 v4l2h264dec` to check if the codec existed and turned out that it wasn't installed.
I then found that the required library was `libgstvideo4linux2.so` based on some searching and found that was contained in the `gstreamer1.0-plugins-good` package which `install.sh` doesn't install.

After installing this package and rebuilding openauto and dash it seems to be working correctly now.

## Expected Result:
The `install.sh` script should either include `gstreamer1.0-plugins-good` or needs to detect the rpi model its running on and selectively install based on the codec that will be used.

Additionally it would be nice if possible that dash logged errors of some form if the codec isn't found.

## Problem-relevant steps to reproduce:

1. Install Raspberry Pi OS Lite on a RPI4
2. Run `sudo apt install --no-install-recommends xserver-xorg-video-all xserver-xorg-input-all xserver-xorg-core xinit x11-xserver-utils`
3. Clone dash to `/home/pi/dash`
4. Run `install.sh` from `/home/pi/dash`

<details> 
  <summary>Optional Steps</summary>

5. Create `/home/pi/.xinitrc`
```sh
#!/usr/bin/env sh
xset -dpms
xset s off
xset s noblank

while [ true ]; do
  sh /home/pi/run.sh
done
```
6. Create `/home/pi/run.sh`
```sh
#!/usr/bin/env sh

/home/pi/dash/bin/ia >> /home/pi/dash/bin/ia.log 2>&1

sleep 1
```
7. Enable console autologin with `sudo raspi-config`.
  `1 System Options` -> `S5 Boot / Auto Login` -> `B2: Console Autologin`

</details>

**Comment by robert5974** [comment]
Yes, the model of Raspberry Pi needs to be I cluded in the CMakeLists file.  I just had this issue because my model of the 4B is new and wasn't included yet.  After adding it to that file, it works.  This has been identified and the resolution has been created.  It is coming soon.

**Comment by robert5974** [comment]
But good work on explaining why it is this way.  Thanks.

**Comment by icecube45** [comment]
@robert5974 cmake pi reversion fix was already merged fwiw.

@ZeroErrors, the install script is based around a Raspbian (or I guess Raspberry Pi OS now) Desktop installation, rather than raspbian lite. Truthfully I'm surprised nothing else is misbehaving on a raspbian lite install. 

Dash actually can't see any gstreamer errors, but gstreamer does do it's own logging - you just need to set the GST_DEBUG environment variable.

I'll wait for others to chime in about reworking the install script to support -lite installations.

**Comment by icecube45** [comment]
@ZeroErrors since you were able to track down where `v4l2h264dec` came from, are you also able to identify the raspbian package for `omxh264dec`?

**Comment by ZeroErrors** [comment]
@icecube45 Sure thing.

So this is based on my RPI4 setup, it seems the `gstreamer1.0-omx-rpi` is the package that provides that codec for me.

However, it looks like there is also another version installed which comes from `gstreamer1.0-omx-generic`. I assume the rpi version is just modified to run better on rpi.

<details> 
  <summary>Steps to reproduce</summary>

1. `gst-inspect-1.0 omxh264dec`
  This gives me the file path `/usr/lib/arm-linux-gnueabihf/gstreamer-1.0/libgstomx.so`
  That file is linked via an alternatives
```
/usr/lib/arm-linux-gnueabihf/gstreamer-1.0/libgstomx.so -> /etc/alternatives/libgstomx-arm-linux-gnueabihf.so
/etc/alternatives/libgstomx-arm-linux-gnueabihf.so -> /usr/lib/arm-linux-gnueabihf/gstreamer-1.0/libgstomx-rpi.so
```

2. `dpkg -S libgstomx-rpi.so`
```
gstreamer1.0-omx-rpi: /usr/lib/arm-linux-gnueabihf/gstreamer-1.0/libgstomx-rpi.so
```
_(Can also use the debian package search for some stuff https://www.debian.org/distrib/packages)_

**Other libgstomx:**
`/usr/lib/arm-linux-gnueabihf/gstreamer-1.0/libgstomx-generic.so` 
```
> dpkg -S /usr/lib/arm-linux-gnueabihf/gstreamer-1.0/libgstomx-generic.so
gstreamer1.0-omx-generic: /usr/lib/arm-linux-gnueabihf/gstreamer-1.0/libgstomx-generic.so
```

</details>

**Comment by icecube45** [comment]
Awesome, thank you (for the package and for the how-to!)
I'll try to verify results on a pi 3 (where we use omx) if I can find mine. If you're able to go a couple days of use more without finding more broken things from -lite install, should be trivial to make sure the appropriate packages are installed on the -lite distro.
